FastAutoAugment/datasets.py

```diff
@@ -0,0 +1,42 @@
+import torchvision
+from collections import defaultdict
+from PIL import Image
+from theconf import Config as C
+
+class CIFAR10_mod(torchvision.datasets.CIFAR10):
+    def __init__(self, root, train=True, transform=None, target_transform=None,
+                 download=False):
+        super().__init__(root, train, transform, target_transform, download)
+        self.hardness_scores = defaultdict(defaultdict)
+        
+    def __getitem__(self, index):
+        """
+        Args:
+            index (int): Index
+
+        Returns:
+            tuple: (image, target) where target is index of the target class.
+        """
+        img, target = self.data[index], self.targets[index]
+        
+        measure = C.get()['hardness']['aug_measure']
+        if measure in self.hardness_scores and index in self.hardness_scores[measure]:
+            hardness_score = self.hardness_scores[measure][index]
+        else:
+            hardness_score = 1
+        # doing this so that it is consistent with all other datasets
+        # to return a PIL Image
+        img = Image.fromarray(img)
+
+        if self.transform is not None:
+            for t in self.transform.transforms:
+                from FastAutoAugment.data import Augmentation
+                if isinstance(t, Augmentation):
+                    img = t(img, hardness_score)
+                else:
+                    img = t(img)
```

Comment:
we want to apply hardness scores to random flip and cutout as well right? this won't do that right?

Reply:
We don't apply hardness to random flip ever. We always keep its probability as 0.5.  We can apply hardness to cutout. I will change this code to pass hardness score to cutout as well.